### PR TITLE
Dev 1723 - Sage Germline Calling

### PR DESCRIPTION
**Pull request by jonbaber**

### CODE REVIEW

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageGermlineCaller.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return VirtualMachineJobDefinition.sageGermlineCalling(bash, resultsDirectory);
+    }
+
+    @Override
+    public boolean shouldRun(final Arguments arguments) {
+        return arguments.runGermlineCaller();
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageGermlineCaller.java:69]: Can we add a new toggle here so we can turn it off without disabling the GATK germline calling?

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageSomaticCaller.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import com.hartwig.pipeline.execution.vm.unix.UnzipToDirectoryCommand;
+import com.hartwig.pipeline.metadata.SomaticRunMetadata;
+import com.hartwig.pipeline.reruns.PersistedDataset;
+import com.hartwig.pipeline.resource.ResourceFiles;
+import com.hartwig.pipeline.stages.SubStageInputOutput;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+
+public class SageSomaticCaller extends SageCaller {
+
+    public static final String NAMESPACE = "sage_somatic";
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageSomaticCaller.java:25]: What impact will this have downstream on reporting, etc? Will we need to do another rerun to make data homogenous again? 

We could limit this impact of this change by not changing this namespace yet and making that a separate ticket.

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageCaller.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                .addFurtherOperations(new AddDatatypeToFile(dataType, Folder.root(), namespace(), filteredOutputFile, metadata.barcode()));
+    }
+
+    @Override
+    public SageOutput skippedOutput(final SomaticRunMetadata metadata) {
+        return SageOutput.builder(namespace()).status(PipelineStatus.SKIPPED).build();
+    }
+
+    @Override
+    public final SageOutput persistedOutput(final SomaticRunMetadata metadata) {
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageCaller.java:77]: Maybe this guy is better only to implement in the somatic caller. Generally "persisted output" is only required for stages with other stages which depend on them, so you could leave this out of sage germline.

**Review by pauldwolfe — CHANGES_REQUESTED** [review]

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageGermlineCaller.java (hunk 1 above)
Comment (jonbaber) [reply cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageGermlineCaller.java]: Done. Added run_sage_germline_caller argument.

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageSomaticCaller.java (hunk 2 above)
Comment (jonbaber) [reply cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageSomaticCaller.java]: Done. Reverted to sage.

File: cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageCaller.java (hunk 3 above)
Comment (jonbaber) [reply cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageCaller.java:77]: Left as discussed in email thread as we will be running GATK and sage germline in parallel for a while.

**Pull request merged**